plugins/action/zos_copy.py

```diff
@@ -293,8 +299,13 @@ def run(self, tmp=None, task_vars=None):
 
     def _copy_to_remote(self, src, is_dir=False, ignore_stderr=False):
         """Copy a file or directory to the remote z/OS system """
-
-        temp_path = "/{0}/{1}/{2}".format(gettempprefix(), _create_temp_path_name(), os.path.basename(src))
+        self.tmp_dir = self._connection._shell._options.get("remote_tmp")
+        rc, stdout, stderr = self._connection.exec_command("cd {0} && pwd".format(self.tmp_dir))
+        if rc > 0:
+            msg = f"Failed to resolve remote temporary directory {self.tmp_dir}. Ensure that the directory exists and user has proper access."
+            return self._exit_action({}, msg, failed=True)
+        self.tmp_dir = stdout.decode("utf-8").replace("\r", "").replace("\n", "")
+        temp_path = "{0}/{1}/{2}".format(self.tmp_dir, _create_temp_path_name(), os.path.basename(src))
```

Comment:
Could this leave us with extra forward slashes (i.e. `/user/tmp//directory/src_basename`) if the user specifies something like this for the tmp dir: `/user/tmp/`?

If that is not an issue, this code is fine, otherwise I suggest using `temp_path = os.path.join(self.tmp_dir, _create_temp_path_name(), os.path.basename(src))`

Reply:
done